packages/sdk-core/src/modules/breadcrumbs/BreadcrumbsManager.ts

```diff
@@ -0,0 +1,118 @@
+
+    /**
+     * Determines if the breadcrumb manager is enabled.
+     */
+    private _enabled = true;
+    private readonly _eventSubscribers: BreadcrumbsEventSubscriber[] = [new ConsoleEventSubscriber()];
+
+    constructor(configuration?: BacktraceBreadcrumbsSettings, options?: BreadcrumbsSetup) {
+        this.breadcrumbsType = configuration?.eventType ?? defaultBreadcurmbType;
+        this.logLevel = configuration?.logLevel ?? defaultBreadcrumbsLogLevel;
+        this.breadcrumbsStorage = options?.storage ?? new InMemoryBreadcrumbsStorage(configuration?.maximumBreadcrumbs);
+        if (options?.subscribers) {
+            this._eventSubscribers.push(...options.subscribers);
+        }
+    }
+
+    public dispose(): void {
+        this._enabled = false;
+        for (const subscriber of this._eventSubscribers) {
+            subscriber.dispose();
+        }
+    }
+
+    public get(): Record<string, number> {
+        return {
+            [this.BREADCRUMB_ATTRIBUTE_NAME]: this.breadcrumbsStorage.lastBreadcrumbId,
+        };
+    }
+
+    public start() {
+        for (const subscriber of this._eventSubscribers) {
+            subscriber.start(this);
+        }
+    }
+
+    public verbose(message: string, attributes?: Record<string, AttributeType> | undefined): boolean {
+        return this.log(message, BreadcrumbLogLevel.Verbose, attributes);
+    }
+    public debug(message: string, attributes?: Record<string, AttributeType> | undefined): boolean {
+        return this.log(message, BreadcrumbLogLevel.Debug, attributes);
+    }
+    public info(message: string, attributes?: Record<string, AttributeType> | undefined): boolean {
+        return this.log(message, BreadcrumbLogLevel.Info, attributes);
+    }
+    public warn(message: string, attributes?: Record<string, AttributeType> | undefined): boolean {
+        return this.log(message, BreadcrumbLogLevel.Warning, attributes);
+    }
+    public error(message: string, attributes?: Record<string, AttributeType> | undefined): boolean {
+        return this.log(message, BreadcrumbLogLevel.Error, attributes);
+    }
+    public log(
+        message: string,
+        level: BreadcrumbLogLevel,
+        attributes?: Record<string, AttributeType> | undefined,
+    ): boolean {
+        return this.addBreadcrumb(message, level, BreadcrumbType.Manual, attributes);
+    }
+
+    public logReport(report: BacktraceReport) {
+        const level = report.data instanceof Error ? BreadcrumbLogLevel.Error : BreadcrumbLogLevel.Warning;
+        return this.addBreadcrumb(report.message, level, BreadcrumbType.System);
+    }
+
+    public addBreadcrumb(
+        message: string,
+        level: BreadcrumbLogLevel,
+        type: BreadcrumbType,
+        attributes?: Record<string, AttributeType> | undefined,
+    ): boolean {
+        if (!this._enabled) {
+            return false;
+        }
+        if ((this.logLevel & level) !== level) {
+            return false;
+        }
```

Comment:
>This is because different logger implementations may have different levels, and it should be up to them to manage to whether add the breadcrumb or not.

I agree, this is why  each event handler has to convert a specific log level to the expected Backtrace log level. Putting filtering options on each event subscriber doesn't make sense - it will create a lot of code that we don't need to have. 


so tldr;

I would prefer to leave it in the breadcrumbs manager and I would prefer to require event subscribers to convert library specific log levels to our log levels.